addons/binding/org.openhab.binding.avmfritz/src/main/java/org/openhab/binding/avmfritz/handler/BoxHandler.java

```diff
@@ -127,45 +130,53 @@ public void dispose() {
      * {@inheritDoc}
      */
     @Override
+    public void setStatusInfo(ThingStatus status, ThingStatusDetail statusDetail, String description) {
+        super.updateStatus(status, statusDetail, description);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public FritzahaWebInterface getWebInterface() {
+        return this.connection;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
     public void addDeviceList(DeviceModel device) {
         try {
             logger.debug("set device model: {}", device);
-            this.deviceList.put(device.getIdentifier(), device);
-            ThingUID thingUID = this.getThingUID(device);
-            Thing thing = this.getThingByUID(thingUID);
+            deviceList.put(device.getIdentifier(), device);
+            ThingUID thingUID = getThingUID(device);
+            Thing thing = getThingByUID(thingUID);
             if (thing != null) {
                 logger.debug("update thing {} with device model: {}", thingUID, device);
-                this.updateThingFromDevice(thing, device);
+                updateThingFromDevice(thing, device);
             }
         } catch (Exception e) {
             logger.error("{}", e.getLocalizedMessage(), e);
         }
     }
 
     /**
-     * {@inheritDoc}
-     */
-    @Override
-    public FritzahaWebInterface getWebInterface() {
-        return this.connection;
-    }
-
-    /**
      * Updates things from device model.
      *
      * @param thing Thing to be updated.
      * @param device Device model with new data.
      */
     private void updateThingFromDevice(Thing thing, DeviceModel device) {
         if (thing == null || device == null) {
-            throw new IllegalArgumentException("thing or device null, cannot perform update");
+            throw new IllegalArgumentException("thing or device is null, cannot perform update");
         }
         if (device.getPresent() == 1) {
             thing.setStatusInfo(new ThingStatusInfo(ThingStatus.ONLINE, ThingStatusDetail.NONE, null));
-            logger.debug("about to update thing {} from device {}", thing.getUID(), device);
             if (device.isTempSensor() && device.getTemperature() != null) {
                 Channel channelTemp = thing.getChannel(CHANNEL_TEMP);
-                updateState(channelTemp.getUID(), new DecimalType(device.getTemperature().getCelsius()));
+                updateState(channelTemp.getUID(),
+                        new DecimalType(TemperatureModel.toCelsius(device.getTemperature().getCelsius())));
```

Comment:
This became strange, see also the temperature class, maybe you should call the helper method inside the getters?